src/components/tools/geometry-tool/geometry-tool.tsx

```diff
@@ -270,36 +272,38 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
           onDragOver={this.handleDragOver}
           onDragLeave={this.handleDragLeave}
           onDrop={this.handleDrop} />,
-      this.renderMenuButton(),
       this.renderRotateHandle()
     ]);
   }
 
-  private renderMenuButton() {
-    const { model, readOnly } = this.props;
+  private renderToolbar() {
     const { board } = this.state;
-    const isSelectedTile = this.stores.ui.isSelectedTile(model);
-    if (!isSelectedTile || readOnly || !board) return;
+    if (!board) return;
+    const { readOnly } = this.props;
+    const content = this.getContent();
 
-    const angleIcon = renderUnicodeCharAsIconElement("∡");
     const selectedObjects = this.getContent().selectedObjects(board);
     const selectedPoints = selectedObjects && selectedObjects.filter(isPoint);
     const selectedPoint = selectedPoints && (selectedPoints.length === 1)
                             ? selectedPoints[0] as JXG.Point : undefined;
     const supportsVertexAngle = selectedPoint && canSupportVertexAngle(selectedPoint);
-    const hasVertexAngle = selectedPoint && !!getVertexAngle(selectedPoint);
-    const vertexAngleText = !hasVertexAngle ? "Show Angle Label" : "Hide Angle Label";
-    const enableLabelAngle = !readOnly && supportsVertexAngle;
+    const hasVertexAngle: boolean = !!selectedPoint && !!getVertexAngle(selectedPoint);
+    const enableVertexAngle: boolean = !readOnly && !!supportsVertexAngle;
+    const enableDelete: boolean = !readOnly && board && content.hasSelection();
+    const enableDuplicate: boolean = !readOnly && board && !!this.getOneSelectedPolygon();
```

Comment:
Also note, I started with all buttons turned on and then started turning things off - hence the disabled instead of enabled props.  Not sure I like it now that I look back, but it's fine for now.